-fsanitize=undefined link error

Building with g++9.3 or clang10 with -fsanitize=undefined, OpenEXRTest is getting a link error:

/usr/bin/ld: CMakeFiles/OpenEXRTest.dir/testExistingStreams.cpp.o:(.data.rel+0x218): undefined reference to `typeinfo for Imf_3_0::TiledRgbaInputFile'

Any clues what might be causing this?

--- a/src/lib/OpenEXR/ImfTiledRgbaFile.h
+++ b/src/lib/OpenEXR/ImfTiledRgbaFile.h
@@ -305,7 +305,7 @@ class IMF_EXPORT_TYPE TiledRgbaOutputFile
 // Tiled RGBA input file
 //
 
-class TiledRgbaInputFile
+class IMF_EXPORT_TYPE TiledRgbaInputFile
 {
   public:
 

